### PR TITLE
feat: refactor `color` and remove `COLOR`

### PR DESCRIPTION
**Pull request by 0618**
## Problem

**Issue number, if available:**

https://github.com/aws-amplify/amplify-backend/issues/1141

## Changes

- remove `color`
- add `error` to `format`
- use `format` to replace `color`

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by 0618** [comment]
Not sure why the new commits don't show up on the PR, but it's in the branch 
image

**Review by sobolk — APPROVED** [review]

File: packages/cli-core/API.md [hunk 1]
```diff
@@ -62,7 +57,7 @@ export class Printer {
     constructor(minimumLogLevel: LogLevel, stdout?: NodeJS.WriteStream, stderr?: NodeJS.WriteStream, refreshRate?: number);
     indicateProgress(message: string, callback: () => Promise<void>): Promise<void>;
     log(message: string, level?: LogLevel, eol?: boolean): void;
-    print: (message: string, colorName?: COLOR) => void;
+    print: (message: string | string[], formatName?: keyof typeof format) => void;
```
Comment (edwardfoyle) [inline packages/cli-core/API.md:60]: this method signature should be `print: (message: string) => void`. The caller can call format methods and pass the result to print. Here's a usage example: https://github.com/aws-amplify/amplify-backend/blob/26cdffd3a9960a610bf5315f83941afea0596f89/packages/create-amplify/src/amplify_project_creator.ts#L57-L58

**Review by edwardfoyle — COMMENTED** [review]
Heading in the right direction

File: packages/cli-core/API.md (hunk 1 above)
Comment (0618) [reply packages/cli-core/API.md:60]: It could, but not accurate. If `print.list` only accept `string[]`, we should take it into consideration.

Comment (edwardfoyle) [reply packages/cli-core/API.md:60]: that would be `printer.print(format.list(elements))`. The format method returns a single string.

Comment (0618) [reply packages/cli-core/API.md:60]: ohhhh, I see what you meant. Done 8cac3740d7

File: packages/cli/src/error_handler.test.ts [hunk 2]
```diff
@@ -35,10 +35,6 @@ void describe('generateCommandFailureHandler', () => {
     assert.equal(mockPrint.mock.callCount(), 1);
     assert.equal(mockShowHelp.mock.callCount(), 1);
     assert.equal(mockExit.mock.callCount(), 1);
-    assert.deepStrictEqual(mockPrint.mock.calls[0].arguments, [
```
Comment (0618) [inline packages/cli/src/error_handler.test.ts:38]: had to remove the assertions because the results are different on Mac and Linux

Comment (edwardfoyle) [reply packages/cli/src/error_handler.test.ts:38]: Could we use `assert.match(mockPrint.mock.calls[0].arguments[0], new RegExp(someMsg)`?

**Review by edwardfoyle — DISMISSED** [review]

**Review by sobolk — APPROVED** [review]

File: .changeset/new-meals-destroy.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'@aws-amplify/cli-core': patch
```
Comment (sobolk) [inline .changeset/new-meals-destroy.md:2]:
```suggestion
'@aws-amplify/cli-core': minor
```

there are api changes !

**Review by sobolk — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**